### ISSUE TITLE
patterndb parser: undocumented feature

### ISSUE THREAD

**Issue by faxm0dem** [balabit/syslog-ng#147]
It _seems_ it is possible to modify a message using _actions_ without creating new ones by omitting the _message_ element.
Is this an undocumented feature, or a bug?

```
#
@version: 3.5
#
source s_file{
    file("/dev/stdin" flags(no-parse)) ;
};
#
parser p_myparser_pdb {
  db_parser(
    file("/tmp/patterndb.xml")
  );
};
#
destination d_file {
   file("/dev/stdout" template("MESSAGE=``${MESSAGE}`` answer=``${answer}``\n") );
};

#
log {
  source(s_file)  ;
  parser(p_myparser_pdb);
  destination(d_file);
};
```

``` xml
<?xml version='1.0' encoding='UTF-8'?>
<patterndb version='4' pub_date='2014-06-16'>
  <ruleset name='myname' id='123456789'>
    <rules>
      <rule id='123' class='myclass' provider='bcn'>
        <patterns>
            <pattern>@ESTRING:greeting: @@ANYSTRING:who@</pattern>
        </patterns>
        <actions>
          <action trigger='match'>
            <values>
              <value  name="answer">hi</value>
            </values>
          </action>
        </actions>
      </rule>
    </rules>
  </ruleset>
</patterndb>
```

This is _somehow_ functional, but doesn't _validate_ the _xsd_.

```
⚡ syslog-ng -Fv
syslog-ng: Error setting capabilities, capability management disabled; error='Operation not permitted'
hello mr oizo
MESSAGE=`hello mr oizo` answer=`hi`
this_aint_gonna_match
MESSAGE=`this_aint_gonna_match` answer=``
```

IMHO this is a useful feature, though maybe not very comprehensive in its "current" form. Maybe this would be more clear:

``` xml
<action>
  <rewrite>
    ...
  </rewrite>
</action>
```

**Comment by ihrwein** [comment]
@bazsi should we mark this as a feature or a bug?

**Comment by ihrwein** [comment]
@bazsi What do you think? I'd expect from an `action` that it creates a new message so I'd consider this as a bug.

**Comment by bazsi** [comment]
I agree its  a bug. In intended actions to generate a separate message. If
you want to enrich the current message it should just be added to the rule
tag.

**Comment by MrAnno** [comment]
This has been fixed in syslog-ng v3.8.1 (#1032, https://github.com/balabit/syslog-ng/commit/82b473ece85c3426bb502f0c55afad4daf2f5938#diff-d9f8b4f62fd3294230ca43a28a2c0cb1R308).

Since then, only the following actions are supported:
- `create-context`
- `message`

`<value>` as an action is not valid anymore:
```
Error parsing pattern database file; filename='pdb.xml', error='pdb.xml:12:1: Unexpected <values> tag, expected <message> or <create-context>'
```

If you want to achieve the same behavior, it can be done with the following cleaner syntax:
```
<?xml version='1.0' encoding='UTF-8'?>
<patterndb version='4' pub_date='2014-06-16'>
  <ruleset name='myname' id='123456789'>
    <rules>
      <rule id='123' class='myclass' provider='bcn'>
        <patterns>
            <pattern>@ESTRING:greeting: @@ANYSTRING:who@</pattern>
        </patterns>
        <values>
          <value  name="answer">hi</value>
        </values>
      </rule>
    </rules>
  </ruleset>
</patterndb>
```
